### DISCUSSION TITLE
Turborepo and public folder caching

### DISCUSSION THREAD

**Discussion by mirshko** [vercel/community#780] [Help] [answered]
Hey friends! 

I'm having some troubles with Turborepo caching with Next on Vercel. 

I'm following the standard Turborepo build command in the docs `cd ../.. && npx turbo run build --filter=APP_NAME`, the build command for this project runs a node script to download images from a JSON file and dumps them into `public/images/_generated` with fs writeFile, a turborepo cache-miss on this project downloads the images and the build ends up with the images in the Output folder when I inspect it, but a cache-hit ends up with the replayed output not having the files in the Output folder. 

Is this an issue with Vercel or my turborepo config?  

The build command for turborepo is: 

```json
"build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**", ".next/**"]
},
```

Since Vercel caches the public folder at build, should I add the public folder here to the "outputs" for turbo repo so that comes as well since the `next build` command doesn't put the public folder inside the '.next' folder

**Comment by tknickman** [comment, marked as answer]
@mirshko I'm assuming by:
> the build ends up with the images in the Output folder

You mean they end up in `public/` after the build? If so, you're correct, and you just need to adjust your outputs as follows:

`"outputs": ["dist/**", ".next/**", "public/**"]`

Right now you're only telling turbo to cache the contents of `.next/` and `dist/` - which doesn't include `public/` so you'll need to explicitly tell turbo you want to cache that directory.

**Reply by tknickman** [reply]
Also be aware that you'll need to make sure turbo knows when to bust this cache with something in the repo (script args / env vars / etc) otherwise you will always get the same images.

**Reply by mirshko** [reply]
So I tried that with `public/images/generated_` either the issue was missing the `/**` or perhaps because the `_generated` folder is in .gitignore to keep it out of the repo it wasn't cached. Any clue here?

**Reply by tknickman** [reply]
Ah yep, that will do it - right now we exclude anything that is .gitignored when calculating hashes. We're working on some better options for this in the future since that's not always a valid assumption (like this in this case!). For example - this change https://github.com/vercel/turborepo/pull/1699 (pending release) will allow you explicitly list files in "inputs" to include in the hash even if they are git ignored.

**Reply by mirshko** [reply]
Ahh word! Well that explains it, would be dope to have stuff about this in the docs! 

Thanks so much for clearing this up!
